### PR TITLE
feat(core): add ngx_brotli module

### PR DESCRIPTION
**Pull request by oowl**
### Summary

add ngx_brotli module to Kong prebuilt Nginx binary. So we can use these configs in kong configure to support brotli compression feature in Kong.

```
nginx_proxy_brotli = "on"
nginx_proxy_brotli_comp_level = 6
nginx_proxy_brotli_types = "text/plain text/css application/json application/x-javascript text/xml application/xml application/xml+rss text/javascript application/javascript text/x-js"
```

### Checklist

- [x] The Pull Request has tests
- [x] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

KAG-2477

### CODE REVIEW

**Comment by locao** [comment]
Is (or will be) there a sister PR for the docs repo?

File: scripts/explain_manifest/fixtures/amazonlinux-2-amd64.txt [hunk 1]
```diff
@@ -200,6 +200,7 @@
   - lua-kong-nginx-module
   - lua-kong-nginx-module/stream
   - lua-resty-events
+  - ngx_brotli
```
Comment (chobits) [inline scripts/explain_manifest/fixtures/amazonlinux-2-amd64.txt:203]: think putting it after the line of `lua-resty-lmdb` looks better as you did in other fixtures/*.txt

Comment (oowl) [reply scripts/explain_manifest/fixtures/amazonlinux-2-amd64.txt]: done

**Comment by oowl** [comment]
> Is (or will be) there a sister PR for the docs repo?

Where can I add brotli compress feature doc? It seems our doc has not described anything related to the nginx inject Nginx  Directives list. https://docs.konghq.com/gateway/latest/reference/nginx-directives/

**Review by chobits — APPROVED** [review]

**Comment by windmgc** [comment]
Could you run a full matrix test build to verify the manifests are all correct?

**Comment by oowl** [comment]
> Could you run a full matrix test build to verify the manifests are all correct? Like the one here [#11629 (comment)](https://github.com/Kong/kong/pull/11629#issuecomment-1733098068)

https://github.com/Kong/kong/actions/runs/7579992609

**Review by fffonion — APPROVED** [review]

**Comment by fffonion** [comment]
merge on CI and CD both pass

File: .requirements [hunk 2]
```diff
@@ -17,3 +17,5 @@ NGX_WASM_MODULE=a7087a37f0d423707366a694630f1e09f4c21728
 WASMER=3.1.1
 WASMTIME=14.0.3
 V8=10.5.18
+
+NGX_BROTLI=25f86f0bac1101b6512135eac5f93c49c63609e3 # v1.0.0rc
```
Comment (dndx) [inline .requirements:21]: This release version is really old, I wonder if we should pin to the `master` commit `a71f9312c2deb28875acc7bacfdd5695a111aa53` instead as it seems to contain a couple of fixes along the way.

Comment (oowl) [reply .requirements]: done!

**Review by dndx — APPROVED** [review]

**Comment by oowl** [comment]
cross compiler ld failed in the master branch, So I switched to the google rc version.

**Comment by oowl** [comment]
All matrix test has been passed, So I revert https://github.com/Kong/kong/pull/12367/commits/ae1564c127dea1827f01127959a6e9b5565ca760

**Comment by locao** [comment]
> Where can I add brotli compress feature doc? It seems our doc has not described anything related to the nginx inject Nginx Directives list. https://docs.konghq.com/gateway/latest/reference/nginx-directives/

It seems you are right, we don't have these features documented anywhere.

I got late to the party, but LGTM 😄🚀

**Pull request merged**